wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,327 @@
+/**
+	DamageModify
+	by Trokkin
+
+	API:
+
+	// Pretty much '(Damage d) -> ...'
+	interface ForDamageCallback
+		run(Damage d)
+
+	// these are called before the damage is dealt
+	// can be called multiple amount of times or not called at all
+	addDamageModifier(ForDamageCallback cb)
+	removeDamageModifier(ForDamageCallback cb)
+	
+	// these are called after the damage is dealt
+	// are called strictly once per damage instance
+	addDamageListener(ForDamageCallback cb)
+	removeDamageListener(ForDamageCallback cb)
+
+	// a short for 'new Damage(this, target, amount, isSpell, id)..modify()..deal()'
+	unit.damage(unit target, real amount, OPTIONAL bool isSpell = true, int id = 0)
+
+	// a short for '= new Damage(this, target, amount, isSpell, id)..saveInstance()..modify()..deal()'
+	unit.returnDamage(unit target, real amount, bool isSpell, int id) -> Damage
+
+	class Damage
+		
+		// Basic damage event info
+		unit source
+		unit target
+		bool spell
+		real initial
+
+		// .id is used for any unique iterations
```

Comment:
"unique iterations"?

Reply:
See examples.
I don't really know how to comment on that field properly, though mostly I assume that id should correspond to ability that caused the damage.

Reply:
See examples.
I don't really know how to comment on that field properly, though mostly I assume that id should correspond to ability that caused the damage.

Reply:
I think you should remove comments that don't make any sense/aren't meaningful